NitroxModel/Helper/NitroxUser.cs

```diff
@@ -59,8 +69,22 @@ public static string LauncherPath
 
         public static string PreferredGamePath
         {
-            get => RegistryEx.Read<string>(PREFERRED_GAMEPATH_REGKEY);
-            set => RegistryEx.Write(PREFERRED_GAMEPATH_REGKEY, value);
+            get {
+                if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows)) {
+                    return RegistryEx.Read<string>(PREFERRED_GAMEPATH_REGKEY);
+                } else {
+                    return linuxConfig.SubnauticaGamePath;
+                }
+                
+            }
+            set {
+                if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows)) {
+                    RegistryEx.Write(PREFERRED_GAMEPATH_REGKEY, value);
+                } else {
+                    linuxConfig.SubnauticaGamePath = value;
+                    linuxConfig.Save();
+                }
+            }
```

Comment:
Since `RegistryEx` is our code, I would prefer we refactor that to make it compatible as a key/value store for Linux (using something on linux that is conventional for key/value config store). Renaming it to something more generic for both OSes. This would reduce the amount of code lines for Linux checking elsewhere.

Reply:
Sounds like a good idea. Will make changes.

Reply:
Should I keep RegistryEx around, and create a new KeyValueConfig class, that uses RegistryEx on Windows and .config on linux?

Reply:
Yes that's also good with me. Up to you. Either rename RegistryEx or wrap it in another API.

Reply:
Should I use System.Text.Json for the .config file? Attempting to use NitroxModel (net472 project) from Nitrox.Launcher (net6.0 project) for testing the .config file approach causes a runtime error: `Could not load file or assembly 'System.Text.Json, Version=7.0.0.1, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51'. The located assembly's manifest definition does not match the assembly reference. (0x80131040)`. Changing NitroxModel's `TargetFramework` to `<TargetFrameworks>net472;net6.0</TargetFrameworks>` fixes this. This change doesn't seem to affect anything else negatively. Using an XMLDocument is not possible, as dictionaries can't be serialized, and adding Newtonsoft.Json could cause headaches, as this is used by the game.

Reply:
Nevermind. Fixed by downgrading System.Text.Json from 7.0.1 to 7.0.0.

Reply:
Addressed in newest commits.